src/groupeddatatable/grouping.jl

```diff
@@ -236,14 +236,29 @@ colwise(sum, groupby(dt, :a))
 ```
 
 """
-colwise(f::Function, d::AbstractDataTable) = Any[vcat(f(d[idx])) for idx in 1:size(d, 2)]
+function colwise(f::Function, d::AbstractDataTable)
+    x = [f(d[i]) for i in 1:ncol(d)]
+    if eltype(x) <: Nullable
+        return NullableArray(x)
+    else
+        return x
+    end
+end
 colwise(f::Function, gd::GroupedDataTable) = map(colwise(f), gd)
 colwise(f::Function) = x -> colwise(f, x)
 colwise(f) = x -> colwise(f, x)
 # apply several functions to each column in a DataTable
-colwise{T<:Function}(fns::Vector{T}, d::AbstractDataTable) =
-    reshape(Any[vcat(f(d[idx])) for f in fns, idx in 1:size(d, 2)],
-            length(fns)*size(d, 2))
+function colwise{T<:Function}(fns::Vector{T}, d::AbstractDataTable)
+    res = Array{AbstractVector}(ncol(d))
+    for i in eachindex(res)
```

Comment:
> Is eachindex really less efficient?

For some reason I thought there were cases where constructing the `UnitRange` directly was more efficient than going through `eachindex`, but I may have seen that in very old code or I could be making it up entirely.

I tend to apply `@inbounds` pretty liberally: basically any time I have to index and know that it can't be out of bounds. If it's not needed here then that's fine.